### ISSUE TITLE
ERROR: appimage_register_in_system : boost::filesystem::permissions: Operation not permitted

### ISSUE THREAD

**Issue by probonopd** [AppImage/libappimage#101]
Getting

`ERROR: appimage_register_in_system : boost::filesystem::permissions: Operation not permitted: "/isodevice/Applications/kdenlive-17.12.0-x86_64.AppImage"` for AppImages which are stored in the read-only location `/isodevice/Applications`.

Why does it want to write there? This needs to be fixed. We cannot assume that AppImages reside on writeable partitions.

Is this only about setting the executable bit? Then it should give a more appropriate message, and only if the file is not already executable.

```
appimaged, continuous build (commit 2e34378), build 108 built on 2019-03-14 17:43:20 UTC
```

**Comment by probonopd** [comment]
KDenlive is not in the menu.

**Comment by azubieta** [comment]
> Is this only about setting the executable bit? Then it should give a more appropriate message, and only if the file is not already executable.

Yes, will take a look at ti.

**Comment by azubieta** [comment]
Fixed on `libappimage:fix_set_execution_permissions_operation_not_permitted`. Mind to try it with `appimaged`?

**Comment by probonopd** [comment]
> Mind to try it with `appimaged`?

[Build](https://travis-ci.org/AppImage/appimaged/builds/529503292) is running :)

**Comment by probonopd** [comment]
Still getting `ERROR: appimage_register_in_system : boost::filesystem::permissions: Operation not permitted: "/isodevice/Applications/Kdenlive-x86_64.AppImage"`. 

```
appimaged, continuous build (commit 8f9d05e), build 119 built on 2019-05-07 21:33:27 UTC
```
How can I know which version of `libappimage` is in there? (Hello [monorepo](https://github.com/AppImage/AppImageKit/issues/931)...)

**Comment by azubieta** [comment]
> How can I know which version of libappimage is in there? 
```
cd lib/libappimage/
git status # this should tell you in which libappimage branch you are 
git fetch
git checkout origin/fix_set_execution_permissions_operation_not_permitted
```

**Comment by probonopd** [comment]
No, I mean in the built result. (I don't build anything locally.)

**Comment by azubieta** [comment]
This should be have the binaries fixed: https://artifacts.assassinate-you.net/artifactory/appimaged/travis-121/appimaged-x86_64.AppImage

https://travis-ci.org/AppImage/appimaged/jobs/529519741

**Comment by azubieta** [comment]
by the way don't merge that branch in appimaged, it should come from libappimage:master once the pr is merged into master.

**Comment by probonopd** [comment]
Ah, it was not fixed yet in `libappimage` master. I see

**Comment by probonopd** [comment]
https://artifacts.assassinate-you.net/artifactory/appimaged/travis-121/appimaged-x86_64.AppImage looks good to me.

**Comment by azubieta** [comment]
fixed in #110, please reopen if the problem persists

**Comment by probonopd** [comment]
Still getting this

```
me@host:~$ '/home/me/Downloads/appimaged-x86_64.AppImage' --version
+appimaged, continuous build (commit 334ff48), build 143 built on 2019-05-10 05:55:51 UTC
(...)
ERROR: appimage_register_in_system : boost::filesystem::permissions: Operation not permitted: "/isodevice/Applications/Kdenlive-x86_64.AppImage"
ERROR: appimage_register_in_system : Unexpected char '_' at 0
ERROR: appimage_register_in_system : boost::filesystem::permissions: Operation not permitted: "/isodevice/Applications/k3b-git20170204-x86_64.AppImage"
ERROR: appimage_get_type : Unknown AppImage format: /isodevice/Applications/Slic3rPEProfiles/print/sp 0.45mm FAST 0.6 nozzle.ini
ERROR: appimage_get_type : Unknown AppImage format: /isodevice/Applications/Slic3rPEProfiles/print/sp 0.35mm FAST 0.6 nozzle.ini
```

Please always confirm using a continuous `appimaged` build from Travis CI build that the changes are actually working. Thanks!

**Comment by probonopd** [comment]
`ERROR: appimage_register_in_system ` is gone, verified in appimaged, continuous build (commit 1c7986a), build 146 built on 2019-05-11 07:44:39 UTC
